pkg/interceptors/clientid/interceptors.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2025 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package clientid
+
+import (
+	"context"
+	"os"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const CGClientID = "cgclientid"
+const CGRequestID = "cgrequestid"
+
+func getClientID() string {
+	e, err := os.Executable()
+	if err != nil {
+		return "unknown"
+	}
+	return e
+}
+
+func AppendClientID(ctx context.Context) context.Context {
```

Comment:
nit: Does this need to be exported? Might make things unnecessarily hard to change down the line as we don't know if others are using this in different contexts.